order-book/src/test/java/com/market/banica/order/book/service/grpc/componentTests/OrderBookComponentIT.java

```diff
@@ -0,0 +1,227 @@
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit.jupiter.SpringJUnitConfig;
+
+import java.io.IOException;
+import java.util.concurrent.Executors;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.verify;
+
+@SpringJUnitConfig
+@SpringBootTest(classes = OrderBookApplication.class, webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@RunWith(JUnit4.class)
+@ActiveProfiles("testOrderBookIT")
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_CLASS)
+class OrderBookComponentIT {
+
+    @SpyBean
+    @Autowired
+    private AuroraClient auroraClient;
+
+    @Autowired
+    private OrderBookService orderBookService;
+
+    @Autowired
+    private TestConfiguration testConfiguration;
+
+    @Rule
+    public GrpcCleanupRule grpcCleanup = new GrpcCleanupRule();
+
+    @LocalServerPort
+    private int grpcMockPort;
+
+    @Value(value = "${product-name}")
+    private String productName;
+
+    @Value(value = "${client-id}")
+    private String clientId;
+
+    @Value(value = "${product-price}")
+    private double productPrice;
+
+    @Value(value = "${market-topic-prefix}")
+    private String orderBookTopicPrefix;
+
+    private ItemOrderBookResponse itemOrderBookResponse;
+
+    private static ManagedChannel channel;
+    private static ManagedChannel channelTwo;
+
+    private String serverName;
+    private String serverNameTwo;
+    private OrderBookServiceGrpc.OrderBookServiceBlockingStub blockingStub;
+    private AuroraServiceGrpc.AuroraServiceStub asynchronousStub;
+
+    @BeforeEach
+    void setupChannel() {
+        serverName = InProcessServerBuilder.generateName();
+        serverNameTwo = InProcessServerBuilder.generateName();
+
+        channel = InProcessChannelBuilder.forName(serverName).directExecutor().build();
+        channelTwo = InProcessChannelBuilder.forName(serverNameTwo).executor(Executors.newSingleThreadExecutor()).build();
+
+        asynchronousStub = AuroraServiceGrpc.newStub(channelTwo);
+        blockingStub = OrderBookServiceGrpc.newBlockingStub(grpcCleanup.register(channel));
+    }
+
+    @Test
+    void getOrderBookItemLayers_Should_ReturnAUnaryResponse() throws IOException {
+        // Arrange
+        ItemOrderBookResponse expected = generateItemOrderBookExpectedResponse();
+
+        grpcCleanup.register(InProcessServerBuilder
+                .forName(serverName).directExecutor().addService(testConfiguration.getGrpcOrderBookServiceItemLayers(expected))
+                .build().start());
+        // Act
```

Comment:
it is a small thing but it is always better to have an empty line before the comments, looks better and is easier for reading

Reply:
Done